bazel/rules_foreign_cc.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/foreign_cc/built_tools/pkgconfig_build.bzl b/foreign_cc/built_tools/pkgconfig_build.bzl
+index 64cb677..d654f67 100644
+--- a/foreign_cc/built_tools/pkgconfig_build.bzl
++++ b/foreign_cc/built_tools/pkgconfig_build.bzl
+@@ -13,7 +13,7 @@ load("//toolchains/native_tools:tool_access.bzl", "get_make_data")
+ def _pkgconfig_tool_impl(ctx):
+     make_data = get_make_data(ctx)
+     script = [
+-        "./configure  --with-internal-glib --prefix=$$INSTALLDIR$$",
++        "CFLAGS=-Wno-int-conversion CXXFLAGS=-Wno-int-conversion ./configure  --with-internal-glib --prefix=$$INSTALLDIR$$",
```

Comment:
wondering if we only want these flags for m1 where its required

Reply:
can we use `select` here?

Reply:
i think so - not sure how that is specified in terms of platform/arch for m1

Reply:
... i guess arch=arm and platform = macos

Reply:
nm - i see https://github.com/envoyproxy/envoy/blob/4a3989af957c9b99683c142bb72a6b4f467b667e/bazel/BUILD#L854-L861

Reply:
> ... here?

also just realized this is a rule so no wrt to `select` no you cant here

not sure exactly how you handle platform detection in a rule

Reply:
https://bazel.build/rules/lib/builtins/ctx#target_platform_has_constraint

```starlark
    if ctx.target_platform_has_constraint(os="mac") and ctx.target_platform_has_constraint(cpu="arm64"):
        args = ["--mac", "--args"]
   else:
       args = []
   args += ["remaining", "args"]
```

Reply:
@zirain if you can address this part im happy to land

i would rather not disable `no-int-conversion` for all platforms - and would need a second opinion before landing

Reply:
update, it's needed on mac intel with latest xcode.

BTW, need merge https://github.com/envoyproxy/envoy/pull/33231 first?

Reply:
ah, i see, thanks for clarifying